services/nvidia-gpu-operator/24.3.1/defaults/cm.yaml

```diff
@@ -33,4 +33,7 @@ data:
       enabled: true
       version: 3.3.5-3.4.1-ubuntu22.04
     validator:
-      version: v24.3.0
+      version: v24.3.0-d2iq.0
```

Comment:
Don't we need to override repository in these similarly to operator?

Reply:
I added it for operator as the mapping wasn't happening. but for validator I think not an issue as the image is also a part of **services/nvidia-gpu-operator/24.3.1/helmrelease/extra-images.txt**.

Reply:
The `extra-images` file defines additional images that cannot be gathered from helm chart itself. Its usually a case when an operator (installed by a helm chart) uses this value to launch additional deployments. So its not directly visible. I think that in this case, if we only provide new version without the repository, the GPU operator will fail when launched.

Here you need to check the GPU operator helm chart and see if repositories for these images can be provided to the operator.

Reply:
I referred to the below one 
https://github.com/NVIDIA/gpu-operator/blob/bfefa4d700783a3e1e4db07bf0e30840818abe96/deployments/gpu-operator/values.yaml#L67

Reply:
https://github.com/NVIDIA/gpu-operator/blob/bfefa4d700783a3e1e4db07bf0e30840818abe96/deployments/gpu-operator/values.yaml#L52-L54 - validator image is searched in the `repository: nvcr.io/nvidia/cloud-native`. If only the `version: v24.3.0-d2iq.0` is changed the config would still try to pull data from `nvcr.io/nvidia/cloud-native`, where the `v24.3.0-d2iq.0` doesn't exist. This config needs to be updated add

```
      repository: ghcr.io/mesosphere/dkp-container-images/nvcr.io/nvidia/cloud-native
      version: v24.3.0-d2iq.0
```

The same applies on the `devicePlugin` section - https://github.com/NVIDIA/gpu-operator/blob/bfefa4d700783a3e1e4db07bf0e30840818abe96/deployments/gpu-operator/values.yaml#L243-L245

Reply:
updated